src/context/data-context.jsx

```diff
@@ -0,0 +1,47 @@
+import { createContext, useContext, useEffect, useReducer } from "react";
+import { initialReducerData, dataReducer } from "../reducer/dataReducer";
+import axios from "axios";
+
+const DataContext = createContext({
+  contextData: initialReducerData,
+  dispatch: () => {},
+});
+
+const DataProvider = ({ children }) => {
+  const [contextData, dispatch] = useReducer(dataReducer, initialReducerData);
+
+  useEffect(() => {
+    (async () => {
+      dispatch({ type: "LOADING" });
+
+      try {
+        const CategoriesResponse = await axios.get("/api/categories");
+
+        dispatch({
+          type: "SUCCESS_CATEGORIES",
+          payload: CategoriesResponse.data.categories,
+        });
+
+        const ProductsResponse = await axios.get("/api/products");
+
+        dispatch({
+          type: "SUCCESS_PRODUCTS",
+          payload: ProductsResponse.data.products,
+        });
+      } catch (error) {
+        dispatch({
+          type: "ERROR",
+          payload: "Error: Something is Wrong",
+        });
+      }
+    })();
```

Comment:
Just a suggestion you can create a custom hook for fetching data from api.

Reply:
I tried it. But got a few errors.
I'll try it once again after completing basic functionalities.